### ISSUE TITLE
after calling AngularFireOfflineDatabase.list() it keeps being subscribed to firebase events even when list ref is no longer in use.

### ISSUE THREAD

**Issue by ciekawy** [adriancarriger/angularfire2-offline#30]
not sure how this should look like yet I can see `setupList` (and `setupObject`) to call `ref.subscribe()` and never `unsubscribe`, also the `listCache` (and `objectCache`) seem to keep cached values forever. As a result I can see `AngularFireOfflineDatabase.setList()` is being called for each internal and external list modification. Probably there should be at least some optional strategies available to automatically/manually clear cache entries and unsubscribe from firebase updates.

Other thing is that for any list modification (i.e. for `child_added`) always all list is being processed which is probably a candidate for separate issue.

**Comment by adriancarriger** [comment]
@ciekawy turns out there's already a [PR for this](https://github.com/adriancarriger/angularfire2-offline/pull/18).. I'm just waiting on some minor changes. Feel free to check it out and leave feedback here.

Also, a more efficient way to modify lists would be nice! 👍
